### ISSUE TITLE
Delphes side-effects when non-existent file

### ISSUE THREAD

**Issue by Sinclert** [diana-hep/madminer#424]
This issue aims to tackle an issue that may arise when using the [DelphesReader](https://github.com/diana-hep/madminer/blob/master/madminer/delphes/delphes_reader.py#L23) object when interacting with [Pythia8](http://home.thep.lu.se/~torbjorn/Pythia.html) created files.

---

### Current situation

MadMiner offers a way of interacting with one of the MadGraph underlying dependencies, [Delphes](https://cp3.irmp.ucl.ac.be/projects/delphes), through the [DelphesReader](https://github.com/diana-hep/madminer/blob/master/madminer/delphes/delphes_reader.py#L23) class. This class offers a way of adding samples to an instantiated object, before running the job.

This sequence of _"adding samples"_ and then _"running Delphes"_ seems to be the norm when it comes to working with Physics workflows, as it can be seen in the [particle physics tutorial](https://cranmer.github.io/madminer-tutorial/tutorial_particle_physics/2b_delphes_level_analysis.html#2.-Run-Delphes).

---

### Issue

The issue is located within the [add_sample](https://github.com/diana-hep/madminer/blob/master/madminer/delphes/delphes_reader.py#L106) function, as it allows to specify file path arguments such as `hepmc_filename` and `lhe_filename`, without checking if they exist. This is a problem that may arise later, when executing the [run_delphes](https://github.com/diana-hep/madminer/blob/master/madminer/delphes/delphes_reader.py#L195) function, as it relies on the previously-added samples.

---

### Proposal

The first action to solve this problem would be to check that both `hepmc_filename` and `lhe_filename` paths of the [DelphesReader.add_sample](https://github.com/diana-hep/madminer/blob/master/madminer/delphes/delphes_reader.py#L106) function exist. Something like:

```python
if not os.path.exists(hepmc_filename):
    raise ValueError("The specified hepmc file does not exists")
if not os.path.exists(lhe_filename):
    raise ValueError("The specified lhe file does not exists")
```

---

**EDIT:**

The root of the confusion came from the use of `.hepmc` Pythia8 sample files instead of their compressed version (`.hepmc.gz`) when calling [DelphesReader.add_sample](https://github.com/diana-hep/madminer/blob/master/madminer/delphes/delphes_reader.py#L106).

I originally thought that the output files from the [MadMiner.run_multiple](https://github.com/diana-hep/madminer/blob/master/madminer/core/madminer.py#L784) invocation were different if the `only_prepare_script` option was used. But it is not the case.

**Comment by johannbrehmer** [comment]
I think these checks are a good idea, though (as you mention in your edit) we should be careful to check for both the compressed and uncompressed version.

Do you have time to open a PR for this?

**Comment by Sinclert** [comment]
Yes.

I created a PR (https://github.com/diana-hep/madminer/pull/426), where I also included validations for the `add_observable` and `add_cut` functions, which receive wildcard strings, which need to be a valid Python expressions.

**Comment by johannbrehmer** [comment]
Thank you! So this is solved with 027fec93e326415180712c47b031791a148c5479
